### ISSUE TITLE
[BUG] [SDW] clock stop errors while removing jack during speaker playback

### ISSUE THREAD

**Issue by plbossart** [thesofproject/linux#1603]
Configuration:

https://github.com/thesofproject/linux/pull/1586

options snd-sof-intel-hda-common sdw_clock_stop_quirks=0x8

speaker-test -Dhw:0,2 -c2 -r48000

insert the jack after 5s and remove it again after 5s

<details> <summary> clock restart errors </summary>

````
[   77.775348] intel-sdw sdw-master-0: intel_resume_runtime start
[   77.775745] intel-sdw sdw-master-0: Clock is stopped
[   77.775758] rt711 sdw:0:25d:711:0: sdw_modify_slave_status: changing status slave 1 status 1 new status 0
[   77.775764] rt711 sdw:0:25d:711:0: sdw_modify_slave_status: initializing completion for Slave 1
[   77.775824] intel-sdw sdw-master-0: Slave status change
[   77.775832] intel-sdw sdw-master-0: Slave 1 ALERT
[   77.775839] intel-sdw sdw-master-0: sdw_handle_slave_status: start
[   77.775846] intel-sdw sdw-master-0: sdw_handle_slave_alerts: start
[   77.775853] rt711 sdw:0:25d:711:0: sdw_modify_slave_status: changing status slave 1 status 0 new status 2
[   77.776588] intel-sdw sdw-master-0: Msg Ack not received
[   77.776603] intel-sdw sdw-master-0: Msg ignored for Slave 1
[   77.776695] intel-sdw sdw-master-0: Msg Ack not received
[   77.776701] intel-sdw sdw-master-0: Msg ignored for Slave 1
[   77.776707] rt711 sdw:0:25d:711:0: Clock Stop prepare failed for slave: -61
[   77.776722] rt711 sdw:0:25d:711:0: clk stop deprep failed:-61
[   77.776728] intel-sdw sdw-master-0: intel_resume_runtime done
[   77.776749] rt711 sdw:0:25d:711:0: rt711_dev_resume start
[   79.782811] rt711 sdw:0:25d:711:0: Initialization not complete, timed out
[   79.782845] rt711 sdw:0:25d:711:0: Failed to resume device: -22
[   79.782853] intel-sdw sdw-master-0: Slave 1 alert handling failed: -22
[   79.782859] intel-sdw sdw-master-0: Updating Slave 1 status
[   79.782866] intel-sdw sdw-master-0: sdw_handle_slave_status: Updating Slave 1 status done
````

</details>

[dmesg_hs_removal_while_speaker_playback.log](https://github.com/thesofproject/linux/files/3934172/dmesg_hs_removal_while_speaker_playback.log)

Something is quite wrong here because the Slave status is ALERT before it reports as attached. It seems like we restart the clock before doing a bus reset...

**Comment by RanderWang** [comment]
@plbossart  yes, I will check this bug. If speak is working, master0 should be powered on. So  clock stop 0 should work.

**Comment by plbossart** [comment]
Fixed by https://github.com/thesofproject/linux/pull/1605, now merged in PR #1586 

Thanks @shumingfan

**Comment by RanderWang** [comment]
@plbossart   It should be fixed by https://github.com/plbossart/sound/pull/12.  1605 can't fix it. Thanks!
